crates/walrus-service/src/client/responses.rs

```diff
@@ -756,3 +766,25 @@ impl NodeHealthOutput {
         }
     }
 }
+
+/// The output of the `read-quilt` command.
+#[derive(Serialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct ReadQuiltOutput {
+    /// The output directory path.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub out: Option<PathBuf>,
+    /// The retrieved blobs.
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub retrieved_blobs: Vec<QuiltStoreBlob<'static>>,
+}
+
+impl ReadQuiltOutput {
+    /// Creates a new [`ReadQuiltOutput`] object.
+    pub fn new(out: Option<PathBuf>, retrieved_blobs: Vec<QuiltStoreBlob<'static>>) -> Self {
+        Self {
+            out,
+            retrieved_blobs,
+        }
+    }
+}
```

Comment:
Suggestion: Don't include the raw blob data in the struct when `out.is_some()` (similar to the `ReadOutput` above).